[MJAVADOC-544] - Changed behaviour of Javadoc for temporary files encoding (options, argfile, ...)

michael-st opened the pull request:
mvn -Prun-its verify was run with JDK9 on windows without errors

michael-st commented:
Due to changes to readFile in b78042ec conflicts were introduced.

I rebased the PR and fixed some style-issues in the tests.

rfscholte commented on src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java:
```diff
@@ -4368,9 +4378,15 @@ private void addCommandLineOptions( Commandline cmd, List<String> arguments, Fil
 
+        /* default to platform encoding */
+        String encoding = null;
+        if ( JAVA_VERSION.compareTo( SINCE_JAVADOC_9 ) >= 0 )
```
Please switch to `isAtLeast()` or `isBefore()`, which are better readable.

rfscholte commented on src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java:
```diff
@@ -4414,9 +4430,15 @@ private void addCommandLineArgFile( Commandline cmd, File javadocOutputDirectory
 
+        /* default to platform encoding */
+        String encoding = null;
+        if ( JAVA_VERSION.compareTo( SINCE_JAVADOC_9 ) >= 0 )
```
Please switch to `isAtLeast()` or `isBefore()`, which are better readable.

rfscholte replied:
Don't forget to squash your commits

michael-st replied:
I addressed your points and squashed the commits.

Ran mvn -Prun-its verify with JDK8 and JDK9 on Windows successfully.

rfscholte commented:
https://builds.apache.org/view/M-R/view/Maven/job/maven-box/job/maven-javadoc-plugin/job/MJAVADOC-544/ we're not there yet.

michael-st commented:
too bad...

The error message occurs when compiling the file "/src/test/resources/unit/argfileumlautencoding-test/argfileumlautencoding/test/AppÃ¤Ã¶Ã¼Ã.java".
The compiler complains about a name mismatch between class "Appäöüß" and the filename "AppÃ¤Ã¶Ã¼Ã.java".
This is a charset-problem, which occurs only in the linux environment.

I tried to reproduce the problem:

The build passes when I execute the maven build in a docker container (image: maven:3-jdk-9-slim) in which the LANG is set to C.UTF-8.
If I change the LANG-environment variable in the container ("export LANG=en_US.ISO-8859-1") - the build fails.

```
FROM maven:3-jdk-9-slim
RUN echo $LANG
> C.UTF-8
RUN apt update && apt install -y git && echo dirty no cleanup of apt
RUN git clone https://github.com/michael-st/maven-javadoc-plugin.git
RUN cd maven-javadoc-plugin
RUN git checkout MJAVADOC-544
RUN mvn -Dtest=org.apache.maven.plugins.javadoc.JavadocReportTest#testArgfileUmlautEncoding clean test
> build passes
RUN export LANG=en_US.ISO-8859-1
RUN echo $LANG
> en_US.ISO-8859-1
RUN mvn -Dtest=org.apache.maven.plugins.javadoc.JavadocReportTest#testArgfileUmlautEncoding clean test
> build fails
```

I tried to convert the file itself and the corresponding .xml-file to ISO-8859-1, but that did not go well either.

At the moment I'm out of ideas and I don't think that I can find a working solution for this (besides setting the LANG to ...UTF-8 in the build environment).

My initial problem was only related to the options file, which would be fixed with this pull request.
For this all tests should pass (hopefully).

So, for this issue, I have removed the special handling for argfile, which would only be needed if someone is using special characters in java class names.
I also modified the commit message to reflect this change.

Is this course of action okay?
